packages/payment-processor/src/payment/index.ts

```diff
@@ -69,44 +117,86 @@ export async function hasSufficientFunds(
   provider?: Provider,
 ): Promise<boolean> {
   const paymentNetwork = getPaymentNetwork(request);
+  if (!paymentNetwork || !supportedNetworks.includes(paymentNetwork)) {
+    throw new UnsupportedNetworkError(paymentNetwork);
+  }
 
-  let ethBalance;
-  switch (paymentNetwork) {
-    case ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_PROXY_CONTRACT: {
-      if (!provider) {
-        provider = getNetworkProvider(request);
-      }
-      ethBalance = await provider.getBalance(address);
-      const balance = await getErc20Balance(request, address, provider);
-      // check ETH for gas, and token for funds transfer
-      return ethBalance.gt(0) && balance.gt(bigNumberify(request.expectedAmount || 0));
-    }
-    case ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT: {
-      if (!provider) {
-        provider = getNetworkProvider(request);
-      }
-      ethBalance = await provider.getBalance(address);
-      const balance = await getErc20Balance(request, address, provider);
-      const feeAmount =
-        request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT].values
-          .feeAmount || 0;
-      // check ETH for gas, and token for funds transfer
-      return (
-        ethBalance.gt(0) && balance.gt(bigNumberify(request.expectedAmount || 0).add(feeAmount))
-      );
+  if (!provider) {
+    provider = getNetworkProvider(request);
+  }
+
+  let feeAmount = new BigNumber(0);
+  if (paymentNetwork === ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT) {
+    feeAmount = request.extensions[paymentNetwork].values.feeAmount || 0;
+  }
+  return isSolvent(
+    address,
+    request.currencyInfo,
+    new BigNumber(request.expectedAmount).add(feeAmount),
```

Comment:
Any reason why we dropped bigNumberify and started using BigNumber instead?

Reply:
Laziness... haha